Detour/Source/DetourNavMeshQuery.cpp

```diff
@@ -1101,6 +1143,13 @@ dtStatus dtNavMeshQuery::findPath(dtPolyRef startRef, dtPolyRef endRef,
 			if (!filter->passFilter(neighbourRef, neighbourTile, neighbourPoly))
 				continue;
 
+			if (reversedSearch && neighbourPoly->getType() == DT_POLYTYPE_OFFMESH_CONNECTION)
+			{
+				const dtOffMeshConnection* con = m_nav->getOffMeshConnectionByRef(neighbourRef);
+				if (!con || !(con->flags & DT_OFFMESH_CON_BIDIR))
+					continue;
+			}
+
```

Comment:
What is the cost of adding this for long path finds? It should be low since it is 100% correctly predicted, but otherwise we can use a template instead.

Reply:
That check will almost certainly incur a memory access (I doubt there are enough registers), so it's probably too high.  Using a template param for `reversedSearch` is a better idea, and I will do that.

Reply:
This inner loop already has many accesses and this is just a stack variable. I doubt it will register.

Reply:
Made it a template parameter as it makes sense as we currently know at compile time what the value should be.  The method is also private, so we don't need to worry about changing the API in the future if we have a new use-case where we don't know if `reversedSearch` should be true or false at compile time.